dotnet-desktop-guide/net/winforms/controls-design/designer-overview.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Visual Studio Design-time overview
+description: Learn about what capabilities are available in Visual Studio for Windows Forms designer support with your custom controls.
+ms.date: 04/20/2023
+ms.topic: overview
+no-loc: ["UserControl", "UserControl1", "UserControlProject", "Label", "Button", "Form", "TextBox"]
+dev_langs:
+- "csharp"
+- "vb"
+f1_keywords: 
+- "UserControl"
+helpviewer_keywords: 
+- "controls [Windows Forms], user controls"
+- "controls [Windows Forms], types of"
+- "composite controls [Windows Forms]"
+- "extended controls [Windows Forms]"
+- "controls [Windows Forms], extended"
+- "user controls [Windows Forms]"
+- "custom controls [Windows Forms]"
+- "controls [Windows Forms], composite"
+---
+
+# Overview about Visual Studio design-time support for custom controls (Windows Forms .NET)
+
+As you've noticed when interacting with the Windows Forms designer, there are many different design-time features offered by the Windows Forms controls. Some of the features offered by the Visual Studio Designer include snap lines, action items, and the property grid. All of these features offer you an easier way to interact and customize a control during design-time. This article is an overview about what kind of support you can add to your custom controls, to make the design-time experience better for the consumers of your controls.
+
+[!INCLUDE [desktop guide under construction](../../includes/desktop-guide-preview-note.md)]
+
+## What's different from .NET Framework
+
+Many basic design elements of custom controls have remained the same from .NET Framework. However, if you use more advanced designer customization features, such as action lists, type converters, custom dialogs, you have some unique scenarios to handle.
+
+Visual Studio is a .NET Framework-based application, and as such, the Visual Designer you see for Windows Forms is also based on .NET Framework. With a .NET Framework project, both the Visual Studio environment and the Windows Forms app being designed run within the same process, **devenv.exe**. This poses a problem when you're working with a Windows Forms .NET (not .NET Framework) app. Both .NET and .NET Framework can't work within the same process. As a result, Windows Forms .NET uses a different designer, the "out-of-process" designer.
+
+The out-of-process designer is a process called **DesignToolsServer.exe**, and is run along-side Visual Studio's **devenv.exe** process. The **DesignToolsServer.exe** process runs in the same version and platform, such as .NET 7 and x64, of .NET that your app is using. When your custom control needs to display UI in the **devenv.exe** your custom control must implement a client-server architecture to facilitate the communication to and from **devenv.exe**. For more information, see [The designer changes since .NET Framework (Windows Forms .NET)](designer-differences-framework.md).
```

Comment:
I'm not sure which is more appropriate here - would "targeting" perhaps be more accurate than "using" for the "of .NET your app is using"?